scvi/model/base/_base_model.py

```diff
@@ -229,16 +345,11 @@ def _validate_anndata(
                 "Input AnnData not setup with scvi-tools. "
                 + "attempting to transfer AnnData setup"
             )
-            self.register_manager(self.adata_manager.transfer_setup(adata))
-        elif (
-            adata_manager.registry[_SOURCE_SCVI_UUID_KEY]
-            != self.adata_manager.registry[_SCVI_UUID_KEY]
-        ):
-            logger.info(
-                "Input AnnData requires setup with AnnData the model was initialized with. "
-                "Attempting to transfer setup with initial AnnData."
+            self._register_manager_for_instance(
+                self.adata_manager.transfer_setup(adata)
             )
-            self.register_manager(self.adata_manager.transfer_setup(adata))
+        else:
+            adata_manager.validate()
```

Comment:
the manager here is the correct one for the instance and this line will replay registration? If so let's comment that